Add dladm_linkprop module

xen0l opened the pull request:
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New Module Pull Request


##### COMPONENT NAME
dladm_linkprop

##### ANSIBLE VERSION
```
2.3
```


##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Module to manage network link interface properties on illumos/Solaris systems.

xen0l committed:
Add dladm_linkprop module

privateip commented on lib/ansible/modules/network/illumos/dladm_linkprop.py:
```diff
@@ -0,0 +1,272 @@
+dladm_linkprop: name=e1000g1 property=mtu state=reset
+'''
+
+RETURN = '''
```
RETURN doc string is required

privateip commented on lib/ansible/modules/network/illumos/dladm_linkprop.py:
```diff
@@ -0,0 +1,272 @@
+
+    module.exit_json(**result)
+
+from ansible.module_utils.basic import *
```
please use explicit imports

privateip commented on lib/ansible/modules/network/illumos/dladm_linkprop.py:
```diff
@@ -0,0 +1,272 @@
+    module.exit_json(**result)
+
+from ansible.module_utils.basic import *
+main()
```
please wrap in `IF` block

privateip reviewed (changes requested):
* missing METADATA
* please consider adding unit tests

xen0l committed:
Add ANSIBLE_METADATA
Document return values
Make imports explicit
Use native YAML syntax in EXAMPLES

xen0l commented:
ready_for_review

xen0l commented:
@privateip I addressed you comments. As for unit tests, they will be shared among all modules to a certain degree (I am also speaking about modules, which are already included with Ansible). For that reason, I would like to add them in the next round if possible.
